### PR TITLE
feat(query): add prefetch functions

### PR DESCRIPTION
**Pull request by georgiev-anton**
## Status

**READY**

## Description

New function for generating [prefetch](https://tanstack.com/query/v4/docs/react/guides/prefetching) hooks for the SSR Next.JS

### CODE REVIEW

**Review by melloware — CHANGES_REQUESTED** [review]
LGTM but don't you need to update the docs and document this new option?

**Comment by georgiev-anton** [comment]
> LGTM but don't you need to update the docs and document this new option?

sorry, I'll do it soon

**Comment by melloware** [comment]
Thanks!

**Comment by georgiev-anton** [comment]
@melloware done

**Comment by melloware** [comment]
looks like the build is failing?

**Comment by georgiev-anton** [comment]
> looks like the build is failing?

fixed

**Review by melloware — APPROVED** [review]

File: packages/query/src/index.ts [hunk 1]
```diff
@@ -844,7 +848,7 @@ const generateQueryImplementation = ({
           )
             return param.destructured;
           return param.name === 'params'
-            ? `{ ${queryParam}: pageParam, ...params }`
+            ? `{...params, ${queryParam}: pageParam }`
```
Comment (melloware) [inline packages/query/src/index.ts]: My question is the same as @anymaniax asked in the other ticket does this change break backwards compatibility?

**Review by melloware — CHANGES_REQUESTED** [review]

File: packages/query/src/index.ts (hunk 1 above)
Comment (georgiev-anton) [reply packages/query/src/index.ts]: I'm thinking of doing this with another pull request.

**Review by melloware — APPROVED** [review]

**Pull request merged**